Transitioned codebase to bubbles/key keybinds

kanielrkirby opened the pull request:
Fixes #64.

- Added `key.go` to handle keybindings with `bubbles/key`.
- Transitioned `tui.go` to use keybindings defined in `key.go`.
- Added `model.help` to handle help messages.
- Overrode the `list.KeyMap.CancelWhileFiltering` to `q`.
- Added `viewport` full help menu, extracted help keys to `key.go`.
- Changed `g` viewport keybinding to be consistent with `list`.
- Extracted `list` overrides into `key.go#SetOverrides`.

guyfedwards commented on internal/commands/tui.go:
```diff
@@ -541,47 +548,13 @@ func Render(items []list.Item, cmds Commands, errors []string) error {
-			),
-		}
-	}
+	ListKeyMap.SetOverrides(&l)
```
I think we will need an `updaetKeybinds()` function that we can call at various points with filter state/list state/num items etc. Set is good for the initial but related to my other comment, the state seems to get confused after transitioning, might be solved by having an update func.

guyfedwards commented on internal/commands/key.go:
```diff
@@ -0,0 +1,171 @@
+		key.WithKeys("?"),
+		key.WithHelp("?", "more"),
+	),
+	CloseFullHelp: key.NewBinding(
```
Don't think this is working as expected, I see `? more` on initial load, but then see `? close help` when the help is open.

guyfedwards commented:
Great work, this cleans everything up considerably, found a bit of state weirdness. 

- Currently seeing an issue where if I go into filter, hit esc to cancel filter, then I am shown a minimal help and keybinds such as `q` or `esc` don't work.
- We might benefit from a function we can call to update the keybinds, something like https://github.com/charmbracelet/bubbles/blob/48dffdd97e4ff8cb3fb16e6414bc8b6d8ae86a4c/list/list.go#L681 to help with state based keybinds outside of the list/viewport state

kanielrkirby commented:
I'll see if I can address those concerns here in the next couple days in my free time.

kanielrkirby commented:
Addressed those issues. Looks like the state issues are caused by assigning `list.KeyMap` values directly. They provide methods `SetHelp` and `SetKeys` for this purpose, which they actually respect.

guyfedwards commented:
Looks great, thanks @kanielrkirby
